api/src/main/java/com/codeforcommunity/rest/subrouter/ProtectedReportRouter.java

```diff
@@ -55,4 +83,43 @@ private void handleGetStewardshipReportRoute(RoutingContext ctx) {
 
         end(ctx.response(), 200, JsonObject.mapFrom(stewardshipReportResponse).toString());
     }
-}
+
+    private void registerGetStewardshipReportCSV(Router router) {
+        Route getStewardshipReportCSVRoute = router.get("/csv/stewardship");
+        getStewardshipReportCSVRoute.handler(this::handleGetStewardshipReportCSVRoute);
+    }
+
+    private void handleGetStewardshipReportCSVRoute(RoutingContext ctx) {
+        JWTData userData = ctx.get("jwt_data");
+
+        Integer previousDays = getValidatedPreviousDaysParam(ctx);
+
+        GetReportCSVRequest getStewardshipReportCSVRequest = new GetReportCSVRequest(previousDays);
+
+        String stewardshipReportCSVResponse = processor.getStewardshipReportCSV(userData, getStewardshipReportCSVRequest);
+
+        end(ctx.response(), 200, stewardshipReportCSVResponse);
+    }
+
+    /**
+     * Return the validated previousDays query parameter from the request URL. The parameter must be an integer greater
+     * than or equal to 0. If the parameter is not given, a value of DEFAULT_PREVIOUS_DAYS is returned.
+     *
+     * @param ctx routing context
+     * @return validated previousDays parameter
+     */
+    private Integer getValidatedPreviousDaysParam(RoutingContext ctx) {
+        Optional<Integer> maybePreviousDays =
+                RestFunctions.getOptionalQueryParam(ctx, PREVIOUS_DAYS_QUERY_PARAM_NAME, Integer::parseInt);
+        Integer previousDays = maybePreviousDays.orElse(null);
+
+        // validate user input
+        if (previousDays == null || System.currentTimeMillis() - TimeUnit.DAYS.toMillis(previousDays) < 0) {
+            previousDays = DEFAULT_PREVIOUS_DAYS;
+        } else if (previousDays < 0) {
```

Comment:
❓ When is this true but `System.currentTimeMillis() - TimeUnit.DAYS.toMillis(previousDays) < 0` isn't? Aren't they both checks for the future?

Reply:
`previousDays < 0` was true when the time was in the future whereas `System.currentTimeMillis() - TimeUnit.DAYS.toMillis(previousDays) < 0` checked if the date wasn't before the Unix epoch. Without `System.currentTimeMillis() - TimeUnit.DAYS.toMillis(previousDays) < 0`, passing in a number like 10000000 as `previousDays` threw a 500 Internal Server error because the date is out or range.

Reply:
Ah, good to know, nice case coverage!